app/src/main/java/com/practicum/playlistmaker/utils/DateUtils.kt

```diff
@@ -7,14 +7,18 @@ import java.util.TimeZone
 
 object DateUtils {
 
-    fun formatTime(trackTimeMillis: Int): String = SimpleDateFormat("mm:ss", Locale.getDefault())
-        .format(trackTimeMillis)
+    fun formatTime(trackTimeMillis: Int?): String = SimpleDateFormat("mm:ss", Locale.getDefault())
+        .format(trackTimeMillis) ?: ""
 
-    fun getYear(str: String): String {
-        val date = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.getDefault()).parse(str)
-        val calendar = Calendar.getInstance(TimeZone.getDefault()).apply {
-            time = date
+    fun getYear(str: String?): String? {
```

Comment:
🍏  Вместо if-else можно попробовать такой синтаксис:
```suggestion
    fun getYear(str: String?): String? =
       str?.let {
            val date = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.getDefault()).parse(it)
            val calendar = Calendar.getInstance(TimeZone.getDefault()).apply {
                time = date
            }
            calendar.get(Calendar.YEAR).toString()
        }
```